code/src/components/LoadingIndicator.js

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import Lottie from "react-lottie";
+import animationData from "../animations/loading";
+import { useSelector } from "react-redux";
+
+export const LoadingIndicator = () => {
+  const loading = useSelector((store) => store.ui.loading);
+
+  const defaultOptions = {
+    loop: true,
+    autoplay: true,
+    animationData,
+    rendererSettings: {
+      preserveAspectRatio: "xMidYMid slice"
+    }
+  };
+
+  return (
+    <>
+      {loading && <Lottie options={defaultOptions} height={400} width={400} />}
+    </>
+  );
+};
```

Comment:
Separate component for Lottie animation, good job! You really like to have multiple components and it makes your code more structured.

Reply:
Thank you :)